test/unit-test/FreeRTOS_DHCPv6/FreeRTOS_DHCPv6_utest.c

```diff
@@ -0,0 +1,4422 @@
+
+#define TEST_DHCPV6_OPTION_SERVER_ID         ( 0x28BADC54000ACD295EB6 )
+static uint8_t ucTestDHCPv6OptionServerID[] = { 0x28, 0xBA, 0xDC, 0x54, 0x00, 0x0A, 0xCD, 0x29, 0x5E, 0xB6 };
+
+const uint8_t ucDefaultMACAddress[ ipMAC_ADDRESS_LENGTH_BYTES ] = { 0xab, 0xcd, 0xef, 0x11, 0x22, 0x33 };
+const IPv6_Address_t xDefaultNetPrefix = { 0x20, 0x01, 0x04, 0x70, 0xEC, 0x54, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
+
+/* Default IPv6 address is set to 2001:0470:EC54::5 */
+const IPv6_Address_t xDefaultIPAddress = { 0x20, 0x01, 0x04, 0x70, 0xEC, 0x54, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x05 };
+const IPv6_Address_t xDNSAddress[ 3 ] =
+{
+    /* 2001:0470:EC54::FF */
+    { 0x20, 0x01, 0x04, 0x70, 0xEC, 0x54, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFF },
+    /* 2001:0470:EC54::FE */
+    { 0x20, 0x01, 0x04, 0x70, 0xEC, 0x54, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFE },
+    /* 2001:0470:EC54::FD */
+    { 0x20, 0x01, 0x04, 0x70, 0xEC, 0x54, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xFE },
+};
+
+Socket_t xStubFreeRTOS_setsockopt_xSocket;
+size_t xStubFreeRTOS_setsockopt_uxOptionLength;
+int32_t xStubFreeRTOS_setsockopt_lOptionName_BitMap;
+FreeRTOS_Socket_t * xStubvSocketBind_pxSocket;
+
+/* The maximum size in single read/write operation. */
+#define TEST_DHCPv6_BIT_OPERATION_MAX_SIZE              ( 64 )
+/* The maximum number of bit operations in a test case. */
+#define TEST_DHCPv6_BIT_OPERATION_MAX_NUM               ( 128 )
+/* The maximum size of debug message of bit operations. */
+#define TEST_DHCPv6_BIT_OPERATION_DEBUG_MSG_MAX_SIZE    ( 64 )
+
+typedef enum eTestDHCPv6BitOperationType
+{
+    eTestDHCPv6BitOperationNone = 0,
+    eTestDHCPv6BitOperationWrite8,
+    eTestDHCPv6BitOperationWrite16,
+    eTestDHCPv6BitOperationWrite32,
+    eTestDHCPv6BitOperationWriteCustom,
+    eTestDHCPv6BitOperationRead8,
+    eTestDHCPv6BitOperationRead16,
+    eTestDHCPv6BitOperationRead32,
+    eTestDHCPv6BitOperationReadCustom,
+    eTestDHCPv6BitOperationReadPeek,
+    eTestDHCPv6BitOperationSetError,
+    eTestDHCPv6BitOperationReturnFalse,
+} eTestDHCPv6BitOperationType_t;
+
+typedef struct xTestDHCPv6BitOperation
+{
+    eTestDHCPv6BitOperationType_t eOperationType;
+    uint32_t ulCustomLength;
+    union operationValue
+    {
+        uint8_t ucVal;
+        uint16_t usVal;
+        uint32_t ulVal;
+        uint8_t ucValCustom[ TEST_DHCPv6_BIT_OPERATION_MAX_SIZE ];
+    } val;
+    uint8_t ucDebugMsg[ TEST_DHCPv6_BIT_OPERATION_DEBUG_MSG_MAX_SIZE ];
+} xTestDHCPv6BitOperation_t;
+
+xTestDHCPv6BitOperation_t xTestDHCPv6BitOperation[ TEST_DHCPv6_BIT_OPERATION_MAX_NUM ];
+uint32_t ulTestDHCPv6BitOperationWriteIndex;
+uint32_t ulTestDHCPv6BitOperationReadIndex;
+
+/* ============================  Unity Fixtures  ============================ */
+
+/*! called before each test case */
+void setUp( void )
```

Comment:
Why are we defining it?

Reply:
This function is called before every test case. That is provided by Unity framework.
Can it can help us to reset all states before running test cases to ensure all test cases are independent.

Reply:
Yes if are doing something it is needed, i think we can remove tearDown( void )